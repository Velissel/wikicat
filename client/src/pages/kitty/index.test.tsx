import React from 'react';
import { render, screen } from '@testing-library/react';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import Kitty from '.';

jest.mock('../../data/loadCatById');
jest.mock('../../utils/api');

const TEST_DATA = {
  breed: {
    weight: {
      imperial: '5 - 10',
      metric: '2 - 5',
    },
    id: 'tang',
    name: 'Turkish Angora',
    cfa_url: 'http://cfa.org/Breeds/BreedsSthruT/TurkishAngora.aspx',
    vetstreet_url: 'http://www.vetstreet.com/cats/turkish-angora',
    vcahospitals_url:
      'https://vcahospitals.com/know-your-pet/cat-breeds/turkish-angora',
    temperament:
      'Affectionate, Agile, Clever, Gentle, Intelligent, Playful, Social',
    origin: 'Turkey',
    country_codes: 'TR',
    country_code: 'TR',
    description:
      'This is a smart and intelligent cat which bonds well with humans. With its affectionate and playful personality the Angora is a top choice for families. The Angora gets along great with other pets in the home, but it will make clear who is in charge, and who the house belongs to',
    life_span: '15 - 18',
    indoor: 0,
    alt_names: 'Ankara',
    adaptability: 5,
    affection_level: 5,
    child_friendly: 4,
    dog_friendly: 5,
    energy_level: 5,
    grooming: 2,
    health_issues: 2,
    intelligence: 5,
    shedding_level: 2,
    social_needs: 5,
    stranger_friendly: 5,
    vocalisation: 3,
    experimental: 0,
    hairless: 0,
    natural: 1,
    rare: 0,
    rex: 0,
    suppressed_tail: 0,
    short_legs: 0,
    wikipedia_url: 'https://en.wikipedia.org/wiki/Turkish_Angora',
    hypoallergenic: 0,
    reference_image_id: '7CGV6WVXq',
  },
  images: [
    'https://cdn2.thecatapi.com/images/DYOqsRy9f.jpg',
    'https://cdn2.thecatapi.com/images/41Fe8q9vU.jpg',
    'https://cdn2.thecatapi.com/images/xTo8K7wb3.jpg',
    'https://cdn2.thecatapi.com/images/58mi0uCwO.jpg',
    'https://cdn2.thecatapi.com/images/cSzaNCgq2.jpg',
    'https://cdn2.thecatapi.com/images/KU_LUntSX.jpg',
    'https://cdn2.thecatapi.com/images/ZGoY_Ivly.jpg',
    'https://cdn2.thecatapi.com/images/Yx3nQTUHu.jpg',
    'https://cdn2.thecatapi.com/images/GlwgzUMan.jpg',
    'https://cdn2.thecatapi.com/images/7CGV6WVXq.jpg',
  ],
};

describe('Kitty page', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render', async () => {
    const testRouter = createMemoryRouter(
      [
        {
          path: '/',
          element: <Kitty />,
          loader: () => {
            return TEST_DATA;
          },
        },
      ],
      {
        initialEntries: ['/'],
        initialIndex: 0,
      },
    );

    const { container } = render(
      <RouterProvider router={testRouter}></RouterProvider>,
    );
    // await waitFor(() =>screen.findByText('Other Photos'));
    await screen.findByText('Other Photos');

    expect(screen.getByTestId('kitty-photos')).toBeInTheDocument();
    expect(screen.getByTestId('kitty-details')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
