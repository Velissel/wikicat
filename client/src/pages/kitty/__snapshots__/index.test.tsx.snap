// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kitty page should render 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="mt-5 row"
    >
      <div
        class="col-md-4 col-sm-6 col-12"
      >
        <img
          class="img-fluid rounded"
          data-testid="header-image"
          src="https://cdn2.thecatapi.com/images/DYOqsRy9f.jpg"
        />
      </div>
      <div
        class="col-md-8 col-sm-6 col-12"
      >
        <h1>
          Turkish Angora
        </h1>
        <p>
          This is a smart and intelligent cat which bonds well with humans. With its affectionate and playful personality the Angora is a top choice for families. The Angora gets along great with other pets in the home, but it will make clear who is in charge, and who the house belongs to
        </p>
        <ol
          class="list-group"
          data-testid="kitty-details"
        >
          <div
            class="list-group-item"
          >
            <div
              class="ms-2 me-auto"
            >
              <div
                class="fw-bold"
              >
                temperament
              </div>
              Affectionate, Agile, Clever, Gentle, Intelligent, Playful, Social
            </div>
          </div>
          <div
            class="list-group-item"
          >
            <div
              class="ms-2 me-auto"
            >
              <div
                class="fw-bold"
              >
                origin
              </div>
              Turkey
            </div>
          </div>
          <div
            class="list-group-item"
          >
            <div
              class="ms-2 me-auto"
            >
              <div
                class="fw-bold"
              >
                life span
              </div>
              15 - 18
            </div>
          </div>
          <li
            class="d-flex justify-content-between align-items-start list-group-item"
          >
            <div
              class="ms-2 me-auto"
            >
              <div
                class="fw-bold"
              >
                adaptability
              </div>
            </div>
            <span
              class="badge rounded-pill bg-primary"
            >
              5
            </span>
          </li>
          <li
            class="d-flex justify-content-between align-items-start list-group-item"
          >
            <div
              class="ms-2 me-auto"
            >
              <div
                class="fw-bold"
              >
                affectionLevel
              </div>
            </div>
            <span
              class="badge rounded-pill bg-primary"
            >
              5
            </span>
          </li>
          <li
            class="d-flex justify-content-between align-items-start list-group-item"
          >
            <div
              class="ms-2 me-auto"
            >
              <div
                class="fw-bold"
              >
                childFriendly
              </div>
            </div>
            <span
              class="badge rounded-pill bg-primary"
            >
              4
            </span>
          </li>
          <li
            class="d-flex justify-content-between align-items-start list-group-item"
          >
            <div
              class="ms-2 me-auto"
            >
              <div
                class="fw-bold"
              >
                grooming
              </div>
            </div>
            <span
              class="badge rounded-pill bg-primary"
            >
              2
            </span>
          </li>
          <li
            class="d-flex justify-content-between align-items-start list-group-item"
          >
            <div
              class="ms-2 me-auto"
            >
              <div
                class="fw-bold"
              >
                intelligence
              </div>
            </div>
            <span
              class="badge rounded-pill bg-primary"
            >
              5
            </span>
          </li>
          <li
            class="d-flex justify-content-between align-items-start list-group-item"
          >
            <div
              class="ms-2 me-auto"
            >
              <div
                class="fw-bold"
              >
                healthIssues
              </div>
            </div>
            <span
              class="badge rounded-pill bg-primary"
            >
              2
            </span>
          </li>
          <li
            class="d-flex justify-content-between align-items-start list-group-item"
          >
            <div
              class="ms-2 me-auto"
            >
              <div
                class="fw-bold"
              >
                socialNeeds
              </div>
            </div>
            <span
              class="badge rounded-pill bg-primary"
            >
              5
            </span>
          </li>
          <li
            class="d-flex justify-content-between align-items-start list-group-item"
          >
            <div
              class="ms-2 me-auto"
            >
              <div
                class="fw-bold"
              >
                strangerFriendly
              </div>
            </div>
            <span
              class="badge rounded-pill bg-primary"
            >
              5
            </span>
          </li>
        </ol>
      </div>
    </div>
    <h2
      class="mt-5 mb-3"
    >
      Other Photos
    </h2>
    <div
      class="row"
      data-testid="kitty-photos"
    >
      <div
        class="col-lg-3 col-md-4 col-sm-4 col-6"
      >
        <img
          class="mb-4 image img-fluid rounded"
          data-testid="img-0"
          src="https://cdn2.thecatapi.com/images/41Fe8q9vU.jpg"
        />
      </div>
      <div
        class="col-lg-3 col-md-4 col-sm-4 col-6"
      >
        <img
          class="mb-4 image img-fluid rounded"
          data-testid="img-1"
          src="https://cdn2.thecatapi.com/images/xTo8K7wb3.jpg"
        />
      </div>
      <div
        class="col-lg-3 col-md-4 col-sm-4 col-6"
      >
        <img
          class="mb-4 image img-fluid rounded"
          data-testid="img-2"
          src="https://cdn2.thecatapi.com/images/58mi0uCwO.jpg"
        />
      </div>
      <div
        class="col-lg-3 col-md-4 col-sm-4 col-6"
      >
        <img
          class="mb-4 image img-fluid rounded"
          data-testid="img-3"
          src="https://cdn2.thecatapi.com/images/cSzaNCgq2.jpg"
        />
      </div>
      <div
        class="col-lg-3 col-md-4 col-sm-4 col-6"
      >
        <img
          class="mb-4 image img-fluid rounded"
          data-testid="img-4"
          src="https://cdn2.thecatapi.com/images/KU_LUntSX.jpg"
        />
      </div>
      <div
        class="col-lg-3 col-md-4 col-sm-4 col-6"
      >
        <img
          class="mb-4 image img-fluid rounded"
          data-testid="img-5"
          src="https://cdn2.thecatapi.com/images/ZGoY_Ivly.jpg"
        />
      </div>
      <div
        class="col-lg-3 col-md-4 col-sm-4 col-6"
      >
        <img
          class="mb-4 image img-fluid rounded"
          data-testid="img-6"
          src="https://cdn2.thecatapi.com/images/Yx3nQTUHu.jpg"
        />
      </div>
      <div
        class="col-lg-3 col-md-4 col-sm-4 col-6"
      >
        <img
          class="mb-4 image img-fluid rounded"
          data-testid="img-7"
          src="https://cdn2.thecatapi.com/images/GlwgzUMan.jpg"
        />
      </div>
      <div
        class="col-lg-3 col-md-4 col-sm-4 col-6"
      >
        <img
          class="mb-4 image img-fluid rounded"
          data-testid="img-8"
          src="https://cdn2.thecatapi.com/images/7CGV6WVXq.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
